modules/apps/portal/portal-spring-extender-impl/src/main/java/com/liferay/portal/spring/extender/internal/LiferayPortalServiceExtension.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.spring.extender.internal;
+
+/**
+ * @author Hai Yu
+ */
+public interface LiferayPortalServiceExtension {
+
+	public default void destroy() {
```

Comment:
@yuhai Why are these methods `default`? I think every sub class should implement these methods.

Reply:
@dantewang, since LiferayServiceExtender#addingBundle() return LiferayPortalServiceExtension and LiferayServiceExtender#_registerLiferayPortalServiceExtension() registered liferayPortalServiceExtension.

Reply:
@yuhai But this doesn't explain the reason you make these 2 methods `default`. If they are implemented/overridden by all the sub classes, they should not be `default`.

Reply:
@dantewang, ok, I will change this later.

Reply:
@dantewang, ok, I will change this later.